### DISCUSSION TITLE
[OP Stack Deployment] Error when upgrading to ecotone

### DISCUSSION THREAD

**Discussion by luleigreat** [ethereum-optimism/developers#319] [Deploying the OP Stack]
### Did you check the documentation?

- [X] Yes

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

I have an op-stack chain start with version v1.7.0, with the following rollup config(version related only):
```
{
  "regolith_time": 0,
  "canyon_time": 0,
}
```
Now I want to upgrade the chain to ecotone, there is no error when I start op-node with `--override.delta`，but errored when I start with `--override.ecotone`:
```
./op-node \
        --l2=ws://localhost:8551 \
        --l2.jwt-secret=./jwt.txt \
        --sequencer.enabled \
        --sequencer.l1-confs=5 \
        --verifier.l1-confs=4 \
        --rollup.config=./rollup.json \
        --rpc.addr=0.0.0.0 \
        --rpc.port=8547 \
        --rpc.enable-admin \
        --p2p.sequencer.key=$GS_SEQUENCER_PRIVATE_KEY \
        --p2p.listen.ip=0.0.0.0 \
        --p2p.listen.tcp=9003 \
        --p2p.listen.udp=9003 \
        --override.delta=1711437381 \
        --override.ecotone=1711441227 \
        --l1=$L1_RPC_URL \
        --l1.beacon=$L1_BECON_URL \
        --l1.rpckind=$L1_RPC_KIND \
        --log.level=info
```

Then I change the op-geth start command with --override.xxx,but still the same error:
```
./geth \
        --datadir ./datadir \
        --http \
        --http.corsdomain="*" \
        --http.vhosts="*" \
        --http.addr=0.0.0.0 \
        --http.api=web3,debug,eth,txpool,net,engine \
        --ws \
        --ws.addr=0.0.0.0 \
        --ws.port=8546 \
        --ws.origins="*" \
        --ws.api=debug,eth,txpool,net,engine \
        --syncmode=full \
        --gcmode=archive \
        --nodiscover \
        --maxpeers=0 \
        --networkid=420698 \
        --authrpc.vhosts="*" \
        --authrpc.addr=0.0.0.0 \
        --authrpc.port=8551 \
        --authrpc.jwtsecret=./jwt.txt \
        --override.cancun=1711441227 \
        --override.canyon=1711441228 \
        --override.ecotone=1711441229 \
        --rollup.disabletxpoolgossip=true
```

### Logs

t=2024-03-26T18:04:55+0800` lvl=warn msg="Derivation process temporary error" attempts=491 err="stage 0 failed resetting: temp: failed to find the L2 Heads to start from: failed to fetch current L2 forkchoice state: failed to find the L2 head block: failed to parse L1 info deposit tx from L2 block: data is unexpected length: 260"

**Comment by nitantchhajed** [comment]
@luleigreat did you manage to upgrade your op-stack chain to ecotone successfully ?

**Reply by luleigreat** [reply]
No, But I reuild it with ecotone successfully.

**Reply by nitantchhajed** [reply]
Can you tell me the steps to achieve ecotone ??

**Reply by luleigreat** [reply]
Yeah, 
Firstly, before run the sequencer, I configured the .envrc file with the following variables:
```
export OP_BATCHER_DATA_AVAILABILITY_TYPE=blobs

export OP_NODE_OVERRIDE_CANYON=1711437382
export OP_NODE_OVERRIDE_ECOTONE=1711437382
export OP_NODE_OVERRIDE_DELTA=1711437382

export GETH_OVERRIDE_CANYON=1711437382
export GETH_OVERRIDE_CANCUN=1711437382
export GETH_OVERRIDE_ECOTONE=171143738
```
Secondly, After run the op-geth, I query the **genesis block hash**:
```
curl -H "Content-Type: application/json" -X POST --data '{"jsonrpc":"2.0","method":"eth_getBlockByNumber","params":["0x0",false],"id":1}' http://localhost:8545
```
and re-write it to the rullup.json, and then I run the op-node, it won't report error.


The resaon why it report genesis-block hash error is the op-node does not calculate the block hash consistently with op-geth.
